### ISSUE TITLE
Spdy concurrent requests

### ISSUE THREAD

**Issue by ifesdjeen** [netty/netty#3314]
Hi, thanks for the awesome project once again.

We're currently experiencing some issues caused when trying to use same channel/connection for sending (tiny) Spdy messages from multiple threads. Despite the fact messages are being put to `write` sequentially, we still get overlapping Stream IDs on a protocol level (e.g. Data Frame and Syn Stream frames can potentially get interleaved under high load).

I have a couple of workarounds (such as inserting a queue for dispatches), but I wanted to ask if anyone else is experiencing same thing, or maybe there are better strategies for handling it.

**Comment by normanmaurer** [comment]
@ifesdjeen can you give more details ?

**Comment by ifesdjeen** [comment]
I've checked it out more closely. Thing is that on most of requests Netty is sending out at least 3 SPDY frames. And ids have to be monotonically increasing. Because of concurrently dispatched frames, sometimes they come out of order to the server. As soon as a smaller frame comes following larger one, things get out of hand.

We've added locking so that stream id generation would be synchronised with the dispatch, which helped us. Another solution was to have thread-local client channels, which worked quite nicely as well. 

So it works for us by now. I'm (still) (slowly) working on a little SPDY guide. Maybe it'd just make sense to describe a protocol and explain that it may make sense to use one of two of mentioned ways.
